_posts/2021-09-01-tests-template.md

```diff
@@ -0,0 +1,185 @@
+---
+layout: post
+title: Jak pisać czytelne testy?
+subtitle: 
+bigimg: /img/posts/testing.png
+tags: [testy, testowanie, testing, clean code, code quality]
+comments: true
+permalink: /2021/09/tests-template
+published: true
+---
+
+Obecnie niewiele osób zaangażowanych w rozwój oprogramowania jest przeciwna testowaniu. 
+Wyzwanie pojawia się w momencie gdy takie testy trzeba napisać. 
+Często podstawą tego problemu nie jest jakaś podświadoma niechęci do testowania czy też 
+brak dostrzegania w tej czynności wartości. Często chodzi o samo pisanie testów: 
+Jak zacząć? Co robić aby takie testy były przydatne oraz zrozumiałe?
+
+Dzisiaj chciałbym Ci pokazać jak rozpocząć pisanie testów w taki sposób aby były one czytelne. 
+Technikę tą możecie stosować niezależnie od tego czy testy tworzone są po implementacji 
+czy może przed (z wykorzystaniem np. [Test-Driven Development](https://www.youtube.com/watch?v=yiPcPJeTbik&t=46s)).
+
+## Zacznijmy od kodu
+Naszym celem jest przetestowanie funkcjonalności, która ma umożliwić dodawanie nowego Apartamentu
+i jej logika zapewnia następujące działanie:
+1. Jeżeli Właściciel nie istnieje to:
+   1. Publikujemy zdarzenie. 
+   2. Zwracamy wartość nie będącą identyfikatorem 
+2. Jeżeli Właściciel oraz Apartament istnieją to:
+   1. Publikujemy zdarzenie.
+   2. Zwracamy identyfikator istniejącego Apartamentu. 
+3. Jeżeli Właściciel istnieje, ale Apartament nie to:
+   1. Tworzymy nowy Apartament. 
+   2. Zwracamy identyfikator nowego Apartamentu.
+
+Poniżej jest kod realizujący opisane wymagania:
+```java
```

Comment:
A jednak bez gista :(

Reply:
@subieslaw  nie wiem jak. Jeżeli masz jakiś przykład to zmodyfikuję :)